### ISSUE TITLE
Boolean type and boolean operators

### ISSUE THREAD

**Issue by gcarcassi** [ControlSystemStudio/cs-studio#81]
Boolean type and boolean/logic operators are implemented in the following way.

VBoolean is a different type as VString is a different type. It can be used as a number with the ValueUtil.valueOf and ValueUtil.displayOf utility functions.

The following operators > < >= <= == != && || ! & | are implemented with the same semantic and precedence as in Java. The only exception is ^, which is implemented as power instead of bitwise XOR. All bitwise operations are also implemented as functions and( , ) or( , ) xor( , ). Bitwise not was purposely not implemented as it it type dependent (i.e. byte/convert to short/not is different than byte/not/convert to short); better use a mask and XOR.

---

Original request:

In principle, I'd like to add boolean operations, but to do that I need to formalize how to handle boolean.

Some people like to think booleans as a number 0 or 1. Some people don't. Java is on the second camp, so if I implement VBoolean as I have done for VInt/VDouble, VBoolean would _not_ be a VNumber.

The problem of making VBoolean a VNumber is that a java Boolean is not a java Number. So, to make it a VNumber, the getValue() would not return a Boolean, but an Integer, and then add a getBoolean() to get the actual value. This would mess up the Scalar interface, which says that getValue() would return the actual type. So:
VBoolean.getValue() -> Boolean satisfies Scalar but can't satisfy VNumber
VBoolean.getValue() -> Integer satisfies VNumber but breaks Scalar

The other option is not to have a boolean at all, and simply use a number, 0 false, everything else true.

I personally gravitate with making VBoolean not a VNumber, and keep things clean.

For operators, there are similar issues. Do we allow numbers in boolean expressions a-la C, or not a-la Java? For example:
=!2
=2?"A":"B"
are allowed or not?

My personal preference would be to keep it a-la Java. Because it's stricter, and because there are enough other Java-ish things already that I think it's better to just keep one way of doing things than saying: "oh, this bit it's like C, this bit is like Java, this bit is like Lisp, ...".

So, to recap, I need input to answer the following two questions:
- Should VBoolean just be VBoolean, and not extend VNumber? (and therefore can't automatically used in additions and other number operations)
- If so, should boolean operators be restricted to that as in Java (so a?b:c should be written as (a!=0)?b:c)?
